### PR TITLE
Dev: Remove the `config` ignore errors.

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #7731

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Config/ForeignCharacters.php [hunk 1]
```diff
@@ -18,6 +18,8 @@ class ForeignCharacters
 {
     /**
      * Without further ado, the list of foreign characters.
+     *
+     * @var array
```
Comment (paulbalandan) [inline system/Config/ForeignCharacters.php:22]:
```suggestion
     * @var array<string, string>
```

File: system/Config/Publisher.php [hunk 2]
```diff
@@ -35,6 +35,8 @@ class Publisher extends BaseConfig
 
     /**
      * Disables Registrars to prevent modules from altering the restrictions.
+     *
+     * @return void
```
Comment (paulbalandan) [inline system/Config/Publisher.php:39]: Since the method is protected, I think you add `: void` directly.

File: system/Config/View.php [hunk 3]
```diff
@@ -33,13 +33,17 @@ class View extends BaseConfig
      *
      * To prevent potential abuse, all filters MUST be defined here
      * in order for them to be available for use within the Parser.
+     *
+     * @var array
```
Comment (paulbalandan) [inline system/Config/View.php:37]: Can you specify what array type?

File: system/Config/View.php [hunk 4]
```diff
@@ -33,13 +33,17 @@ class View extends BaseConfig
...
      */
     public $filters = [];
 
     /**
      * Parser Plugins provide a way to extend the functionality provided
      * by the core Parser by creating aliases that will be replaced with
      * any callable. Can be single or tag pair.
+     *
+     * @var array
      */
```
Comment (paulbalandan) [inline system/Config/View.php:47]: Can you specify what array type?

Comment (ping-yee) [reply system/Config/View.php:47]: I have added it.

File: system/Config/View.php (hunk 3 above)
Comment (ping-yee) [reply system/Config/View.php]: I have added it.

File: system/Config/ForeignCharacters.php (hunk 1 above)
Comment (ping-yee) [reply system/Config/ForeignCharacters.php]: Done.

File: system/Config/Publisher.php (hunk 2 above)
Comment (paulbalandan) [reply system/Config/Publisher.php]: Sorry, I mean here is final protected

File: system/Config/View.php [hunk 5]
```diff
@@ -33,13 +33,17 @@ class View extends BaseConfig
      *
      * To prevent potential abuse, all filters MUST be defined here
      * in order for them to be available for use within the Parser.
+     *
+     * @var array<string>
```
Comment (paulbalandan) [inline system/Config/View.php:37]:
```suggestion
     * @var array<string, string>
     * @phpstan-var array<string, callable-string>
```

Comment (paulbalandan) [reply system/Config/View.php:37]: Please do the same with `$plugins`, `$corePlugins`, and `$coreFilters`.

File: system/Config/ForeignCharacters.php [hunk 6]
```diff
@@ -18,6 +18,8 @@ class ForeignCharacters
 {
     /**
      * Without further ado, the list of foreign characters.
```
Comment (paulbalandan) [inline system/Config/ForeignCharacters.php:20]: While you're here, please remove this useless comment.

**Comment by MGatner** [comment]
Paul's review looks thorough - skipping but let me know if you need me.

File: system/Config/View.php (hunk 5 above)
Comment (ping-yee) [reply system/Config/View.php:37]:
```log
------ ---------------------------------------------------------------------------------------------- 
  Line   View.php
 ------ ----------------------------------------------------------------------------------------------
  40     Property CodeIgniter\Config\View::$filters type has no signature specified for callable.
  50     Property CodeIgniter\Config\View::$plugins type has no signature specified for callable.
  58     Property CodeIgniter\Config\View::$coreFilters type has no signature specified for callable.
  88     Property CodeIgniter\Config\View::$corePlugins type has no signature specified for callable.
 ------ ----------------------------------------------------------------------------------------------
```

But I get these errors after following the revise.

Comment (paulbalandan) [reply system/Config/View.php:37]: You added `callable-string` and phpstan complains?

Comment (ping-yee) [reply system/Config/View.php:37]: image

Yes, I actually added it.
Show as above.

Comment (paulbalandan) [reply system/Config/View.php:37]: Okay. Please add them as is. Then add the new errors to baseline by: `vendor/bin/phpstan analyse --generate-baseline phpstan-baseline.php`. I'll do the corrections to phpstan.

File: system/Config/ForeignCharacters.php (hunk 6 above)
Comment (ping-yee) [reply system/Config/ForeignCharacters.php]: Done.

**Review by paulbalandan — APPROVED** [review]

**Review by kenjis — APPROVED** [review]

**Pull request merged**